bazel/dependencies/dropbear/0001-override-authorized-keys.patch

```diff
@@ -0,0 +1,113 @@
+@@ -458,22 +458,30 @@ static int checkpubkey(const char* keyalgo, unsigned int keyalgolen,
+ 		dropbear_exit("Failed to set euid");
+ 	}
+ #endif
+-	/* check file permissions, also whether file exists */
+-	if (checkpubkeyperms() == DROPBEAR_FAILURE) {
+-		TRACE(("bad authorized_keys permissions, or file doesn't exist"))
+-	} else {
+-		/* we don't need to check pw and pw_dir for validity, since
+-		 * its been done in checkpubkeyperms. */
+-		len = strlen(ses.authstate.pw_dir);
+-		/* allocate max required pathname storage,
+-		 * = path + "/.ssh/authorized_keys" + '\0' = pathlen + 22 */
+-		filename = m_malloc(len + 22);
+-		snprintf(filename, len + 22, "%s/.ssh/authorized_keys",
+-					ses.authstate.pw_dir);
+-
+-		authfile = fopen(filename, "r");
++	if (svr_opts.authorized_keys_file) {
++		authfile = fopen(svr_opts.authorized_keys_file, "r");
+ 		if (!authfile) {
+-			TRACE(("checkpubkey: failed opening %s: %s", filename, strerror(errno)))
++			TRACE(("checkpubkey: failed flag configured authorized keys file %s: %s",
++				svr_opts.authorized_keys_file, strerror(errno)))
++		}
++	} else {
++		/* check file permissions, also whether file exists */
++		if (checkpubkeyperms() == DROPBEAR_FAILURE) {
++			TRACE(("bad authorized_keys permissions, or file doesn't exist"))
++		} else {
++			/* we don't need to check pw and pw_dir for validity, since
++			 * its been done in checkpubkeyperms. */
++			len = strlen(ses.authstate.pw_dir);
++			/* allocate max required pathname storage,
++			 * = path + "/.ssh/authorized_keys" + '\0' = pathlen + 22 */
++			filename = m_malloc(len + 22);
++			snprintf(filename, len + 22, "%s/.ssh/authorized_keys",
++						ses.authstate.pw_dir);
++
++			authfile = fopen(filename, "r");
++			if (!authfile) {
++				TRACE(("checkpubkey: failed opening %s: %s", filename, strerror(errno)))
++			}
+ 		}
+ 	}
+ #if DROPBEAR_SVR_MULTIUSER
+diff --git svr-runopts.c svr-runopts.c
+index 6870f62..ff105dd 100644
+--- svr-runopts.c
++++ svr-runopts.c
+@@ -110,6 +110,8 @@ static void printhelp(const char * progname) {
+                                         "-A <authplugin>[,<options>]\n"
+                                         "               Enable external public key auth through <authplugin>\n"
+ #endif
++                                        "-O /path/to/authorized_keys_file\n"
++                                        "               Forces dropbear to always use this authorized keys file\n"
+ 					"-V    Version\n"
+ #if DEBUG_TRACE
+ 					"-v    verbose (repeat for more verbose)\n"
+@@ -147,6 +149,7 @@ void svr_getopts(int argc, char ** argv) {
+ #if DROPBEAR_PLUGIN
+         char* pubkey_plugin = NULL;
+ #endif
++        char* authorized_keys_file = NULL;
+ 
+ 
+ 	/* see printhelp() for options */
+@@ -180,6 +183,7 @@ void svr_getopts(int argc, char ** argv) {
+         svr_opts.pubkey_plugin = NULL;
+         svr_opts.pubkey_plugin_options = NULL;
+ #endif
++        svr_opts.authorized_keys_file = NULL;
```

Comment:
More tabs vs spaces

Reply:
good catch, tried to fix as much as possible (dropbear itself seems inconsistent in a few places?)